lib/swoosh/adapters/mailgun.ex

```diff
@@ -86,10 +88,23 @@ defmodule Swoosh.Adapters.Mailgun do
 
   defp prepare_attachments(body, %{attachments: []}), do: body
   defp prepare_attachments(body, %{attachments: attachments}) do
-    Map.put(body, :attachments, Enum.map(attachments, &prepare_file(&1)))
+    normal_attachments = Enum.filter(attachments, fn(%Swoosh.Attachment{type: type}) -> type == :attachment end)
+    inline_attachments = Enum.filter(attachments, fn(%Swoosh.Attachment{type: type}) -> type == :inline end)
```

Comment:
- `Enum.split_with`
- Matching struct type in private functions is not necessary

Reply:
Agreed. Thanks for recommending `Enum.split_with`. I'll fix this on my side.